tests/Integration/Foundation/ExceptionHandlerTest.php

```diff
@@ -151,4 +154,77 @@ public static function exitCodesProvider()
         yield 'Throw exception' => [[Fixtures\Providers\ThrowUncaughtExceptionServiceProvider::class], false];
         yield 'Do not throw exception' => [[Fixtures\Providers\ThrowExceptionServiceProvider::class], true];
     }
+
+    public function testUnauthenticatedWebUsersAreRedirectedToLogin()
+    {
+        Route::get('test-route', fn () => throw new AuthenticationException);
+        Route::get('login', fn () => ':)')->name('login');
+
+        $this->get('test-route')
+            ->assertStatus(302)
+            ->assertRedirect('login');
+    }
+
+    public function testUnauthenticatedUsersAreNotRedirectedIfLoginDoesntExist()
+    {
+        Route::get('test-route', fn () => throw new AuthenticationException);
+
+        $this->assertFalse(Route::has('login'));
+
+        $this->get('test-route')
+            ->assertStatus(401)
+            ->assertSeeText('Unauthenticated.');
```

Comment:
When would an application not have a login route, but yet still have routes protected by auth that would throw an authentication exception like this?

Reply:
When you're using any other auth mechanism, e.g. token in the headers or query.